pkgs/top-level/all-packages.nix

```diff
@@ -10293,7 +10293,9 @@ in
   };
 
   stanchion = callPackage ../servers/nosql/riak-cs/stanchion.nix {
+    inherit (darwin.apple_sdk.frameworks) Carbon Cocoa;
     erlang = erlang_basho_R16B02;
+    pam = lib.optional stdenv.isLinux [ pam ];
```

Comment:
This will result in `{ pam = [ <drv> ]; ... }`, you don't have to do anything special for pam because the darwin derivation won't try to use it anymore.